lib/Dialects/LinalgTransform/Transforms/TransformInterpreter.cpp

```diff
@@ -546,26 +668,73 @@ findTransformTarget(OpTy transformOp, ModuleOp module,
   return applyPatternsOnce(module, std::move(patterns));
 }
 
+template <typename OpTy>
+static std::enable_if_t<!llvm::is_detected<has_target_matcher_t, OpTy>::value,
+                FailureOr<SmallVector<Operation *>>>
+matchUsingPDL(OpTy transformOp, ModuleOp module) {
+  assert(false && "matching with PDL on an unsupported op");
+}
+
+/// Returns the operations that `transformOp` should be run on. The
+/// `transformOp` is expected to have the TargetableTransformOpTrait. Its target
+/// may be either an SSA value that corresponds to a list of ops produced by
+/// other transforms and stored in `operations` or a list of ops that match the
+/// PDL pattern specified by its name.
+template <typename OpTy>
+static FailureOr<SmallVector<Operation *>>
+findTransformTarget(OpTy transformOp, ModuleOp module,
+                    const TransformOpMapping &operations) {
+  if (Value target = transformOp.target())
+    return operations.lookup(target);
+
+  return matchUsingPDL(transformOp, module);
+}
+
+template <typename ConfigOpTy>
+static void removeCurrentTarget(ConfigOpTy configOp,
+                                TransformOpMapping &operations) {
+  // Since there is only allowed use of the value in the transformation dialect,
+  // we can remove it from the mapping after processing its only user. This
+  // ensures we don't accidentally keep pointers to operations that may have
+  // been deleted by the current transformation.
+  if (!configOp.target())
+    return;
+  Value target = configOp.target();
+  assert(target.hasOneUse());
```

Comment:
The assert is already present in the `erase` operation.
Consider adding a more meaningful error message ?

Reply:
You are probably confusing this with erasing the operation itself (`Operation::erase`), rather than removing the value from the dense map called `operations`, that does indeed contain an assertion but for it having _zero_ uses, not one. This assert reflects what we have started to enforce in the verifier recently.

Reply:
ah yes .. I read this as `builder.eraseOp` for some reason.
Please disregard.